clients/include/testing_sbmv.hpp

```diff
@@ -0,0 +1,278 @@
+        rocblas_sbmv<T>(handle, uplo, N, K, &alpha, dA, lda, nullptr, incx, &beta, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_sbmv<T>(handle, uplo, N, K, &alpha, dA, lda, dx, incx, nullptr, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_sbmv<T>(handle, uplo, N, K, &alpha, dA, lda, dx, incx, &beta, nullptr, incy),
+        rocblas_status_invalid_pointer);
+}
+
+template <typename T>
+void testing_sbmv(const Arguments& arg)
+{
+    setenv("HCC_ENABLE_PRINTF", "1", 1);
+
+    rocblas_int N    = arg.N;
+    rocblas_int lda  = arg.lda;
+    rocblas_int K    = arg.K;
+    rocblas_int incx = arg.incx;
+    rocblas_int incy = arg.incy;
+
+    host_vector<T> alpha(1);
+    host_vector<T> beta(1);
+    alpha[0] = arg.alpha;
+    beta[0]  = arg.beta;
+
+    rocblas_fill uplo = char2rocblas_fill(arg.uplo);
+
+    size_t abs_incx = incx >= 0 ? incx : -incx;
+    size_t abs_incy = incy >= 0 ? incy : -incy;
+
+    size_t size_A = size_t(lda) * N;
+    size_t size_X = size_t(N) * abs_incx;
+    size_t size_Y = size_t(N) * abs_incy;
+
+    rocblas_local_handle handle;
+
+    bool bad_uplo = (uplo != rocblas_fill_lower && uplo != rocblas_fill_upper);
+
+    // argument sanity check before allocating invalid memory
+    if(N <= 0 || lda < 0 || K < 0 || !incx || !incy || bad_uplo)
+    {
+        static const size_t safe_size = 100;
+        device_vector<T>    dA(safe_size);
+        device_vector<T>    dx(safe_size);
+        device_vector<T>    dy(safe_size);
+        if(!dA || !dx || !dy)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        rocblas_status status
+            = rocblas_sbmv<T>(handle, uplo, N, K, alpha, dA, lda, dx, incx, beta, dy, incy);
+        if(bad_uplo)
+            EXPECT_ROCBLAS_STATUS(status, rocblas_status_invalid_value);
+        else
+            EXPECT_ROCBLAS_STATUS(status,
+                                  N < 0 || K < 0 || lda < 0 || !incx || !incy
+                                      ? rocblas_status_invalid_size
+                                      : rocblas_status_success);
+        return;
+    }
+
+    // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
+    device_vector<T> d_alpha(1);
+    device_vector<T> d_beta(1);
+
+    host_vector<T> hA(size_A);
+    host_vector<T> hx(size_X);
+    host_vector<T> hy(size_Y);
+    host_vector<T> hy2(size_Y);
+    host_vector<T> hg(size_Y); // gold standard
+
+    double gpu_time_used, cpu_time_used;
+    double rocblas_gflops, cblas_gflops, rocblas_bandwidth;
+    double h_error, d_error;
+
+    device_vector<T> dA(size_A);
+    device_vector<T> dx(size_X);
+    device_vector<T> dy(size_Y);
+    if(!dA || !dx || !dy)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    rocblas_seedrand();
+    rocblas_init_symmetric<T>(hA, N, lda);
```

Comment:
This is being treated as a banded matrix, so there's no use using a special initialization, the symmetry won't be recognized.

If this were treated as a regular matrix, I still don't think we should be initializing it as a symmetric matrix in test code. For symv, for example, we could be accessing elements on the wrong side of the diagonal and still get the correct result.
I think a better way to do it is to just initialize the entire matrix with random data as we do usually, then if we happen to be accessing data on the wrong side of the diagonal, this should be caught by the test.

Reply:
Yes will go with a random fill, before I found the blis functions I was calling gemm to test I'll remove all those.   Clearing half but maybe clear half to nan would be better.